src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -106,36 +108,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Returns the task objects having the ids contained in the array.
+   * Returns the task objects having the ids contained in the array nested into a wrapper
+   * class that also contains the durations.
+   * tasksIds and tasksDurations must have the same length.
    */
-  List<Task> getSelectedTasks(
-      String[] tasksIds, TasksClientAdapter tasksClientAdapter, String tasksListId) {
-    List<Task> tasks = new ArrayList<>();
+  List<ExtendedTask> getSelectedTasksExtended(
+      String[] tasksIds,  String[] tasksDurations, TasksClientAdapter tasksClientAdapter, String tasksListId) {
+    assert tasksIds.length == tasksDurations.length : "tasksIds and tasksDurations have different length";
 
-    for (String id : tasksIds) {
+    List<ExtendedTask> extendedTasks = new ArrayList<>();
+    for (int i = 0; i < tasksIds.length; i++) {
       try {
-        Task task = tasksClientAdapter.getTask(tasksListId, id);
-        tasks.add(task);
+        Task task = tasksClientAdapter.getTask(tasksListId, tasksIds[i]);
+        extendedTasks.add(
+            new ExtendedTask(task, minsToMillis(tasksDurations[i])));
       } catch (IOException IoException) {
```

Comment:
nit: don't include the type in the variable name here since it's already typed... just calling it `exception` or `e` is fine.
if you want to leave it like this, the first letter should be lower case.